pkg/apis/core/v1beta1/helper/helper.go

```diff
@@ -1407,3 +1412,9 @@ func GetFailureToleranceType(shoot *gardencorev1beta1.Shoot) *gardencorev1beta1.
 func SeedWantsManagedIngress(seed *gardencorev1beta1.Seed) bool {
 	return seed.Spec.DNS.Provider != nil && seed.Spec.Ingress != nil && seed.Spec.Ingress.Controller.Kind == v1beta1constants.IngressKindNginx
 }
+
+// IsTopologyAwareRoutingEnabled returns whether the topology aware routing is enabled for the given Shoot running on given Seed.
+// Topology-aware routing is enabled when the corresponding Seed setting is enabled and the Shoot has a multi-zonal control plane.
+func IsTopologyAwareRoutingEnabled(seed *gardencorev1beta1.Seed, shoot *gardencorev1beta1.Shoot) bool {
```

Comment:
Maybe call the function `RouteTopologyAwareForShoot` or similar - the current name is a big misleading when reading it since this is nothing that you explicitly enable for the shoot. It is rather depending on certain characteristics (multi-zonal) and certain settings for the respective seed.

Reply:
Agree. I now changed it to `IsTopologyAwareRoutingForShootControlPlaneEnabled`. Feel free to comment if you have better suggestions.